Pass scope to OIDC_IDTOKEN_PROCESSING_HOOK

suutari-ai opened the pull request:
The ID token processing hook might want to add claims to the ID token
conditionally based on the scope parameter.  Therefore it would be very
useful to provide the scope parameter to the processing hook.

This is a backward incompatible change, but IMHO it's worth it.
Updating the client code requires only adding the scope parameter to the
hook function (or **kwargs might be a good idea too).  Without this
change it requires a nasty hack to get the scope, e.g. something like
this:

https://github.com/suutari-ai/tunnistamo/blob/bb5bfde43a77373ccc636f458ca696817733f1ae/oidc_apis/id_token.py

wojtek-fliposports commented:
I was thinking about this too, to be honest. 
Maybe we change documentation in way that this hook function should have signature similar to this:

```
def token_processing_hook(id_token, user, token, request, **kwargs):
```
`kwargs` for future compatibility.

For right now I'm thinking about passing `request` object and whole `models.Token` (if is created for every type of flows - can't remember right now)

suutari-ai commented:
@wojtek-fliposports I updated this, as you suggested. Now also the token and the request are passed to the hook function. Also updated the documentation.

This should be ready for review now. @juanifioren What do you think?

suutari-ai commented:
> Btw, why the need of passing scope? having token.scope. 

Hmm.. yes. I thought that it's possible for the requested scope to be different than the token scope for some cases, but now that I recheck the code, it seems that I was wrong there. I checked all create_id_token calls and they all send in a `scope` value that is equal to `token.scope` of the `token` sent in the same call.

Should I create a PR to remove the scope param from the arguments?

suutari-ai commented:
@juanifioren I created a PR to remove the scope parameter: #245

suutari-ai commented:
> Removing this will prdoce reverse incompatibility

Incompatibility compared to what? Any version of this lib with the scope pram was never released AFAIK, so it should be safe to remove it.

wojtek-fliposports commented:
With anyone hwho has been using processing hooks to modify process 

 

From: Tuomas Suutari <notifications@github.com>
Reply-To: juanifioren/django-oidc-provider <reply@reply.github.com>
Date: Thursday, 31 May 2018 at 10:50
To: juanifioren/django-oidc-provider <django-oidc-provider@noreply.github.com>
Cc: Wojciech Bartosiak <wojtek@flipsports.com>, Mention <mention@noreply.github.com>
Subject: Re: [juanifioren/django-oidc-provider] Pass scope, token and request to OIDC_IDTOKEN_PROCESSING_HOOK (#194)

 

Removing this will prdoce reverse incompatibility

Incompatibility compared to what? Any version of this lib with the scope pram was never released AFAIK, so it should be safe to remove it.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

juanifioren commented:
and what about using `**kwargs` instead of positional arguments? @suutari-ai @wojtek-fliposports

suutari-ai commented:
@juanifioren The code is already using kwargs when calling the processing hook, so the order of those arguments in the processing hook definition shouldn't matter: https://github.com/juanifioren/django-oidc-provider/blob/develop/oidc_provider/lib/utils/token.py#L65-L67

wojtek-fliposports commented:
+1 for this

 

From: Juan Ignacio Fiorentino <notifications@github.com>
Reply-To: juanifioren/django-oidc-provider <reply@reply.github.com>
Date: Thursday, 31 May 2018 at 19:00
To: juanifioren/django-oidc-provider <django-oidc-provider@noreply.github.com>
Cc: Wojciech Bartosiak <wojtek@flipsports.com>, Mention <mention@noreply.github.com>
Subject: Re: [juanifioren/django-oidc-provider] Pass scope, token and request to OIDC_IDTOKEN_PROCESSING_HOOK (#194)

 

and what about using **kwargs instead of positional arguments? @suutari-ai @wojtek-fliposports

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.
